syntax='proto3';

message SampleMessage {
    message FirstMessage {
        int64 first = 1;
    }

    message SecondMessage {
        int64 second = 1;
        int64 third = 2;
    }

	oneof test_oneof {
		FirstMessage first_message = 2;
		SecondMessage second_message = 3;
	}

	enum MessageType {
		First = 0;
		Second = 1;
	}

	MessageType message_type = 1;
}
message SampleMessage2 {
  oneof test_oneof {
    string name = 4;
    string sub_message = 9;
  }
}
//protoc proto-one-of.proto --cpp_out=.
//g++-7 proto-one-of.cpp proto-one-of.pb.cc -lprotobuf
